app/javascript/packs/form-steps-wait.js

```diff
@@ -0,0 +1,76 @@
+import { loadPolyfills } from '@18f/identity-polyfill';
+
+/**
+ * @typedef FormStepsWaitElements
+ *
+ * @prop {HTMLFormElement} form
+ */
+
+/**
+ * @typedef FormStepsWaitOptions
+ *
+ * @prop {number} pollIntervalMs Poll interval.
+ * @prop {string} waitStepPath URL path to wait step, used in polling.
+ */
+
+/** @type {FormStepsWaitOptions} */
+const DEFAULT_OPTIONS = {
+  pollIntervalMs: 3000,
+  waitStepPath: `${window.location.pathname}_wait`,
+};
+
+class FormStepsWait {
+  constructor(form) {
+    /** @type {FormStepsWaitElements} */
+    this.elements = { form };
+
+    this.options = {
+      ...DEFAULT_OPTIONS,
+      ...this.elements.form.dataset,
+    };
+
+    this.options.pollIntervalMs = Number(this.options.pollIntervalMs);
+
+    this.bindEvents();
+  }
+
+  bindEvents() {
+    this.elements.form.addEventListener('submit', (event) => this.handleSubmit(event));
+  }
+
+  /**
+   * @param {Event} event Form submit event.
+   */
+  async handleSubmit(event) {
+    event.preventDefault();
+
+    const { form } = this.elements;
+    const { action, method } = form;
+
+    await window.fetch(action, {
+      method,
+      body: new window.FormData(form),
+      redirect: 'manual',
+    });
+
+    setTimeout(() => this.poll(), this.options.pollIntervalMs);
+  }
+
+  async poll() {
+    const { waitStepPath, pollIntervalMs } = this.options;
+    const response = await window.fetch(waitStepPath, { redirect: 'manual' });
+    if (!response.ok) {
+      window.location.reload();
+    } else if (response.redirected && new URL(response.url).pathname !== waitStepPath) {
+      window.location.href = response.url;
+    } else {
```

Comment:
A "redirected" response is in 300 range, and since [`.ok` only checks 200 range](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) I think the `response.redirected` check needs to be the first case, right?

Reply:
> A "redirected" response is in 300 range, and since [`.ok` only checks 200 range](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) I think the `response.redirected` check needs to be the first case, right?

Hm, interesting. What you're saying makes sense, though in my testing it appeared to work just fine as-is. I'll have to double-check.

Reply:
> A "redirected" response is in 300 range, and since [`.ok` only checks 200 range](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) I think the `response.redirected` check needs to be the first case, right?

So, funny enough, what I think was happening is that when the redirect happened, while it was technically considered as a `!response.ok`, the refresh that was triggered in that condition was enough to bump the user to the next step, given that the server-side `FormSteps` will automatically force the user to the last completed step.

Reply:
Ok thanks for checking!